pkg/storage/tsdb/backend/filesystem/bucket_client.go

```diff
@@ -0,0 +1,12 @@
+package filesystem
+
+import (
+	"github.com/go-kit/kit/log"
+	"github.com/thanos-io/thanos/pkg/objstore"
+	"github.com/thanos-io/thanos/pkg/objstore/filesystem"
+)
+
+// NewBucketClient creates a new filesystem bucket client
+func NewBucketClient(cfg Config, name string, logger log.Logger) (objstore.Bucket, error) {
```

Comment:
I would leave out unused parameters completely. Or name them `_` to indicate that they are unused.